src/stylize.ts

```diff
@@ -0,0 +1,98 @@
+import {Channel} from 'vega-lite/src/channel';
+import {Type} from 'vega-lite/src/type';
+
+import {QueryConfig} from './config';
+import {isEnumSpec} from './enumspec';
+import {SpecQueryModel} from './model';
+import {EncodingQuery, ScaleQuery} from './query/encoding';
+import {Schema} from './schema';
+import {Dict} from './util';
+
+export function stylize(answerSet: SpecQueryModel[], schema: Schema, opt: QueryConfig): SpecQueryModel[] {
+
+  answerSet = answerSet.map(function(specM) {
+
+    if (opt.smallBandSizeForHighCardinalityOrFacet) {
+      specM = smallBandSizeForHighCardinalityOrFacet(specM, schema);
+     }
+
+    if (opt.nominalColorScaleForHighCardinality) {
+      specM = nominalColorScaleForHighCardinality(specM, schema);
+    }
+    return specM;
+  });
+
+  return answerSet;
+}
+
+let encQIndex: Dict<EncodingQuery> = {};
+
+export function smallBandSizeForHighCardinalityOrFacet(specM: SpecQueryModel, schema: Schema): SpecQueryModel {
+ [Channel.ROW, Channel.Y, Channel.COLUMN, Channel.X].forEach((channel) => {
+        encQIndex[channel] = specM.getEncodingQueryByChannel(channel);
+      });
+
+  const yEncQ = encQIndex[Channel.Y];
+  if (yEncQ !== undefined) {
```

Comment:
Shouldn't that be on the line with if(yEncQ.scale) since we aren't able to evaluate 
(yEncQ.scale as ScaleQuery).type in the event that yEncQ.scale is undefined? (can't get property type of undefined). I've put it on lines 49 and 67 in my most recent push.